desc/protoparse/lexer.go

```diff
@@ -131,7 +131,7 @@ func (l *protoLex) Lex(lval *protoSymType) (code int) {
 			return _ERROR
 		}
 
-		if c == '\n' {
+		if c == '\n' || c == '\r' {
```

Comment:
I don't think `\r` is ever used by itself as a line ending. It's usually either `\n` or, on Windows, both: `\r\n`.

So if we see Windows-style line endings, this code will compute the wrong line number (because it gets incremented once for the `\r` and again for the `\n`).

I think the `\r` should just be ignored:

```go
if c == '\n' {
    l.colNo = 0
    l.lineNo++
    continue
} else if c == '\r' {
    continue
}
```

Reply:
Another possibility might be to leave this as it is, but also to remember the previous token. So if you see a `\r` followed by an `\n` (or vice versa), ignore the second one.

So if you did come across a file that had only `\r`, it would consider it a line ending. But that way it wouldn't double-increment the line number of for files with Windows line-ending conventions.

Reply:
\r as a line ending is only used on older mac's I would think it's safe to expect to not see it.